### PR TITLE
Migrate pkg/kubelet/cm/topologymanager to structure logging

### PR DESCRIPTION
**Pull request by knabben**
#### What type of PR is this?
/kind feature
/sig instrumentation
/sig node

#### What this PR does / why we need it:
Migrate `pkg/kubelet/cm/topologymanager/`

#### Which issue(s) this PR fixes:

Rels https://github.com/kubernetes/kubernetes/issues/98976

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/cm/topologymanager/topology_manager.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ var _ Manager = &manager{}
 
 // NewManager creates a new TopologyManager based on provided policy and scope
 func NewManager(topology []cadvisorapi.Node, topologyPolicyName string, topologyScopeName string) (Manager, error) {
-	klog.Infof("[topologymanager] Creating topology manager with %s policy per %s scope", topologyPolicyName, topologyScopeName)
+	klog.InfoS("[topologymanager] Creating topology manager with policy per scope", "topology", topologyPolicyName, "scope", topologyScopeName)
```
Comment (shubheksha) [inline pkg/kubelet/cm/topologymanager/topology_manager.go:120]:
```suggestion
	klog.InfoS("[topologymanager] Creating topology manager with policy per scope", "topologyPolicyName", topologyPolicyName, "topologyScopeName", topologyScopeName)
```

File: pkg/kubelet/cm/topologymanager/scope_pod.go [hunk 2]
```diff
@@ -73,14 +73,14 @@ func (s *podScope) accumulateProvidersHints(pod *v1.Pod) []map[string][]Topology
 		// Get the TopologyHints for a Pod from a provider.
 		hints := provider.GetPodTopologyHints(pod)
 		providersHints = append(providersHints, hints)
-		klog.Infof("[topologymanager] TopologyHints for pod '%v': %v", format.Pod(pod), hints)
+		klog.InfoS("[topologymanager] TopologyHints", "hints", hints, "pod", format.Pod(pod))
```
Comment (shubheksha) [inline pkg/kubelet/cm/topologymanager/scope_pod.go:76]:
```suggestion
		klog.InfoS("[topologymanager] TopologyHints", "hints", hints, "pod", klog.KObj(pod))
```

**Review by shubheksha — CHANGES_REQUESTED** [review]

File: pkg/kubelet/cm/topologymanager/fake_topology_manager.go [hunk 3]
```diff
@@ -27,31 +27,31 @@ type fakeManager struct{}
 
 //NewFakeManager returns an instance of FakeManager
 func NewFakeManager() Manager {
-	klog.Infof("[fake topologymanager] NewFakeManager")
+	klog.InfoS("[fake topologymanager] NewFakeManager")
```
Comment (ehashman) [inline pkg/kubelet/cm/topologymanager/fake_topology_manager.go:30]:
```suggestion
	klog.InfoS("NewFakeManager")
```

Please remove log prefixes

File: pkg/kubelet/cm/topologymanager/fake_topology_manager.go [hunk 4]
```diff
@@ -27,31 +27,31 @@ type fakeManager struct{}
...
 }
 
 func (m *fakeManager) AddHintProvider(h HintProvider) {
-	klog.Infof("[fake topologymanager] AddHintProvider HintProvider:  %v", h)
+	klog.InfoS("[fake topologymanager] AddHintProvider", "hintProvider", h)
 }
 
 func (m *fakeManager) AddContainer(pod *v1.Pod, containerID string) error {
-	klog.Infof("[fake topologymanager] AddContainer  pod: %v container id:  %v", format.Pod(pod), containerID)
+	klog.InfoS("[fake topologymanager] AddContainer", "pod", format.Pod(pod), "containerId", containerID)
```
Comment (ehashman) [inline pkg/kubelet/cm/topologymanager/fake_topology_manager.go:44]:
```suggestion
	klog.InfoS("AddContainer", "pod", klog.KObj(pod), "containerID", containerID)
```

File: pkg/kubelet/cm/topologymanager/fake_topology_manager.go [hunk 5]
```diff
@@ -27,31 +27,31 @@ type fakeManager struct{}
...
 	return nil
 }
 
 func (m *fakeManager) RemoveContainer(containerID string) error {
-	klog.Infof("[fake topologymanager] RemoveContainer container id:  %v", containerID)
+	klog.InfoS("[fake topologymanager] RemoveContainer", "containerId", containerID)
```
Comment (ehashman) [inline pkg/kubelet/cm/topologymanager/fake_topology_manager.go:49]:
```suggestion
	klog.InfoS("RemoveContainer", "containerID", containerID)
```

File: pkg/kubelet/cm/topologymanager/scope.go [hunk 6]
```diff
@@ -108,7 +108,7 @@ func (s *scope) RemoveContainer(containerID string) error {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v", containerID)
+	klog.InfoS("[topologymanager] RemoveContainer", "containerId", containerID)
```
Comment (ehashman) [inline pkg/kubelet/cm/topologymanager/scope.go:111]:
```suggestion
	klog.InfoS("RemoveContainer", "containerID", containerID)
```

**Review by ehashman — CHANGES_REQUESTED** [review]

**Review by kikisdeliveryservice — DISMISSED** [review]
this is a user facing change, please update description so this gets a release note. thanks!

File: pkg/kubelet/cm/topologymanager/scope_container.go [hunk 7]
```diff
@@ -50,12 +49,12 @@ func (s *containerScope) Admit(pod *v1.Pod) lifecycle.PodAdmitResult {
 
 	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
 		bestHint, admit := s.calculateAffinity(pod, &container)
-		klog.Infof("[topologymanager] Best TopologyHint for (pod: %v container: %v): %v", format.Pod(pod), container.Name, bestHint)
+		klog.InfoS("Best TopologyHint", "bestHint", bestHint, "pod", klog.KObj(pod), "containerName", container.Name, bestHint)
```
Comment (kikisdeliveryservice) [inline pkg/kubelet/cm/topologymanager/scope_container.go:52]: an extra bestHint at the end here ;)

File: pkg/kubelet/cm/topologymanager/scope_container.go [hunk 8]
```diff
@@ -50,12 +49,12 @@ func (s *containerScope) Admit(pod *v1.Pod) lifecycle.PodAdmitResult {
...
 
 		if !admit {
 			return topologyAffinityError()
 		}
-		klog.Infof("[topologymanager] Topology Affinity for (pod: %v container: %v): %v", format.Pod(pod), container.Name, bestHint)
+		klog.InfoS("Topology Affinity", "bestHint", bestHint, "pod", klog.KObj(pod), "containerName", container.Name, bestHint)
```
Comment (kikisdeliveryservice) [inline pkg/kubelet/cm/topologymanager/scope_container.go:57]: extra bestHint

**Review by kikisdeliveryservice — COMMENTED** [review]
caught one thing going thru

**Review by kikisdeliveryservice — COMMENTED** [review]
re-reviewed and all requested changes made. thank you!

/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**